### DISCUSSION TITLE
Modelica.Fluid.Vessels gives error when called.

### DISCUSSION THREAD

**Discussion by 90PlusWorks** [OpenModelica/OpenModelica#11178] [Q&A]
I'm calling this from my code:
Modelica.Fluid.Vessels.ClosedVolume volume(redeclare package Medium = engineAir, V = 0.1 * Modelica.Constants.pi / 4 * 0.1 ^ 2, nPorts = 2, use_portsData=false) annotation(
    Placement(visible = true, transformation(origin = {10, 170}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));

and it gives the following error:
[2] 13:52:49 Translation Error
[Modelica.Fluid.Vessels: 253:9-254:130]: [Non-array modification ‘false‘ for array component ‘fixed‘, possibly due to missing ‘each‘.](omeditmessagesbrowser:///Modelica.Fluid.Vessels?lineNumber=253)

In line 252-3 of Vessels, this is the line:
        // Note: should use fluidLevel_start - portsData.height
        Real[nPorts] s(each start = fluidLevel_max)
        "Curve parameters for port flows vs. port pressures";

Since nPorts = 2, and "each" is not missing, what other error could do this?

**Comment by perost** [comment]
The error message refers to the `fixed` attribute, but it's not set anywhere in the code you've shown. It's possible that it's set somewhere else and the compiler is confused about the actual line number, but we'd need a complete model to say for certain what the issue is.
